ecommerce/api_test.py

```diff
@@ -0,0 +1,46 @@
+"""
+Test for ecommerce functions
+"""
+from django.http.response import Http404
+
+from courses.factories import CourseRunFactory
+from dashboard.factories import ProgramEnrollmentFactory
+from ecommerce.api import get_purchasable_course_run
+from ecommerce.factories import CoursePriceFactory
+from search.base import ESTestCase
+
+
+# pylint: disable=no-self-use
+class PurchasableTests(ESTestCase):
+    """
+    Tests for get_purchasable_courses
+    """
+    def setUp(self):
+        """
+        Create common objects
+        """
+        super(PurchasableTests, self).setUp()
+
+        self.course_run = CourseRunFactory.create(course__program__live=True)
+        # Also creates a user
+        self.program_enrollment = ProgramEnrollmentFactory.create(program=self.course_run.course.program)
+        self.user = self.program_enrollment.user
+        CoursePriceFactory.create(course_run=self.course_run, is_valid=True)
+
+    def test_success(self):
+        """
+        A course run which is live, and has a price, was not already purchased, and
+        where the user is enrolled in the program, should be purchasable
+        """
+        assert get_purchasable_course_run(self.course_run.edx_course_key, self.user) == self.course_run
+
+    def test_not_live(self):
+        """
+        Make sure get_purchasable_course_run raises an Http404 if the program is not live
+        """
+        program = self.course_run.course.program
+        program.live = False
+        program.save()
+
+        with self.assertRaises(Http404):
+            get_purchasable_course_run(self.course_run.edx_course_key, self.user)
```

Comment:
also, we should assume that the other functions work? (I am fine with that if you say so)